src/serve_style.js

```diff
@@ -42,33 +61,53 @@ export const serve_style = {
       }
       // mapbox-gl-js viewer cannot handle sprite urls with query
       if (styleJSON_.sprite) {
-        styleJSON_.sprite = fixUrl(req, styleJSON_.sprite, item.publicUrl);
+        if (Array.isArray(styleJSON_.sprite)) {
+          styleJSON_.sprite.forEach((spriteItem) => {
+            spriteItem.url = fixUrl(req, spriteItem.url, item.publicUrl);
+          });
+        } else {
+          styleJSON_.sprite = fixUrl(req, styleJSON_.sprite, item.publicUrl);
+        }
       }
       if (styleJSON_.glyphs) {
         styleJSON_.glyphs = fixUrl(req, styleJSON_.glyphs, item.publicUrl);
       }
       return res.send(styleJSON_);
     });
 
-    app.get('/:id/sprite:scale(@[23]x)?.:format([\\w]+)', (req, res, next) => {
-      const item = repo[req.params.id];
-      if (!item || !item.spritePath) {
-        return res.sendStatus(404);
-      }
-      const scale = req.params.scale;
-      const format = req.params.format;
-      const filename = `${item.spritePath + (scale || '')}.${format}`;
-      return fs.readFile(filename, (err, data) => {
-        if (err) {
-          console.log('Sprite load error:', filename);
-          return res.sendStatus(404);
+    app.get(
+      '/:id/sprite(/:spriteID)?:scale(@[23]x)?.:format([\\w]+)',
+      (req, res, next) => {
+        const { spriteID = 'default', id } = req.params;
+        const scale = allowedSpriteScales(req.params.scale) || '';
```

Comment:
super minor thing, you can move the `|| ''` in to make it the default:
```suggestion
        const scale = allowedSpriteScales(req.params.scale, { defaultValue: '' });
```

Reply:
when I try this, it scale ends up 'undefined' , so it ends up putting that into the file name. I see this at the console

`Sprite load error: C:\Users\andrew.EIRI\Desktop\test_data\styles\klokantech-basic\spriteundefined.png`